### ISSUE TITLE
Fix remapper plus logger termination

### ISSUE THREAD

**Issue by lrapetti** [robotology/wearables#127]
We have observed that launching an application with the `IWearRemapper` plus the `IWearLogger`, we were not able to terminate it correctly. In particular, we observed that the logger was able to finish the `close()` and reach the end of the `destructor` (so it seems to be closing fine), but the remapper didn't close.

Probably the problem is not at all related to the logger and can happen even in different situation with the remapper, but this should be verified.

cc @kouroshD @prashanthr05 @Yeshasvitvs

**Comment by Yeshasvitvs** [comment]
@lrapetti if you can, please add the termination log details. Thanks.

**Comment by lrapetti** [comment]
I think this problem was fixed, do you remember something @prashanthr05 ?

**Comment by prashanthr05** [comment]
@lrapetti I remember you telling me that you had verified that the termination problem was not related to logger. At that time, you were continuing to verify the remapper.

**Comment by lrapetti** [comment]
> @lrapetti I remember you telling me that you had verified that the termination problem was not related to logger. At that time, you were continuing to verify the remapper.

Yes, actually now I remember that the logger was saving the data so it was basically reaching the termination phase. I will try to investigate again and add more information here.
